### PR TITLE
Add acquire & release to Mutex

### PR DESCRIPTION
**Pull request by BalmungSan**
Yes, we all know using `acquire` & `release` is error-prone compared to `lock`.
However, sometimes you need to be able to split the steps; like here: https://github.com/BalmungSan/programs-as-values/blob/main/src/main/scala/lambda/io/priority/HighLowPriorityRunner.scala#L26-L27

### CODE REVIEW

File: std/shared/src/main/scala/cats/effect/std/Mutex.scala [hunk 1]
```diff
@@ -48,6 +48,28 @@ import java.util.concurrent.atomic.AtomicBoolean
...
+  /**
+   * Acquires the lock.
+   *
+   * @note
+   *   In general prefer to use [[lock]] which returns a `Resource` that ensures the lock is
+   *   properly released.
+   * @see
+   *   [[lock]] [[release]]
+   */
+  def acquire: F[Unit]
```
Comment (armanbilge) [inline std/shared/src/main/scala/cats/effect/std/Mutex.scala:60]: Unfortunately we can't add abstract methods backwards-compatibly.

I feel like there was a rationale for why we didn't add these APIs to `Mutex`, related to typical usage patterns. Meanwhile `Semaphore` offers what you want.

Comment (armanbilge) [reply std/shared/src/main/scala/cats/effect/std/Mutex.scala:60]: Found it:

> if you need code that has a more complicated pattern of locks/unlocks, it's probably not a mutex

https://discord.com/channels/632277896739946517/632278585700384799/1019261725394096230

Comment (BalmungSan) [reply std/shared/src/main/scala/cats/effect/std/Mutex.scala:60]: Yeah, actually, thinking about it, I don't think you need `acquire` and `release` for a `Mutex`

Comment (SystemFw) [reply std/shared/src/main/scala/cats/effect/std/Mutex.scala:60]: It's good that you found that because I was about to repeat it :)

**Pull request closed without merging**